src/System.Memory/tests/AllocationHelper.cs

```diff
@@ -0,0 +1,50 @@
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.SpanTests
+{
+    /// <summary>
+    /// This class is used in testing functions that are allocating significantly large blocks
+    /// of memory that could conceivably cause the machine to go OOM if more than one of the
+    /// tests were run at the same time. This class will block any large allocation call until
+    /// any prior tests have released the memory back to the system.
+    /// </summary>
+    static class AllocationHelper
+    {
+        private static readonly Mutex MemoryLock = new Mutex();
+        private static readonly TimeSpan WaitTimeout = TimeSpan.FromSeconds(120);
+
+        public static bool TryAllocNative(IntPtr size, out IntPtr memory)
+        {
+            memory = IntPtr.Zero;
+
+            if (!MemoryLock.WaitOne(WaitTimeout))
+                return false;
+
+            try
+            {
+                memory = Marshal.AllocHGlobal(size);
+            }
+            catch (OutOfMemoryException)
+            {
+                memory = IntPtr.Zero;
+                MemoryLock.ReleaseMutex();
```

Comment:
No. I don't want to release the Mutex unless there is an exception. In the success case, we hold the mutex and release it when the memory is released. It make sure that we only have one allocation outstanding at a time.